test/jdk/java/nio/channels/DatagramChannel/AdaptorMulticasting.java

```diff
@@ -421,8 +427,22 @@ static void testSendReceive(MulticastSocket s, InetAddress group) throws IOExcep
 
         // receive message
         s.setSoTimeout(0);
-        p = new DatagramPacket(new byte[1024], 100);
-        s.receive(p);
+        do {
+            p = new DatagramPacket(new byte[1024], 100);
+            s.receive(p);
+            if (getPort(p.getSocketAddress()) == getPort(s.getLocalSocketAddress())) {
+                String str = new String(p.getData(), p.getOffset(), p.getLength(), "UTF-8");
+                if (Arrays.equals(p.getData(), p.getOffset(), p.getLength(), message, 0, message.length)) {
+                    System.out.format("Got expected message \"%s\" from %s%n", str, p.getSocketAddress());
+                    break;
+                }
+                System.out.println("Unexpected message received. Expected: " + msg);
+                System.out.println("Received message doesn't match - skipping: " + str);
+            } else {
+                System.out.println("Unexpected message received. Expected message from: " + s.getLocalAddress());
+                System.out.println("Received message sender doesn't match - skipping: " + p.getSocketAddress());
+            }
+        } while(true);
```

Comment:
`do { ... } while(true);` looks a bit odd here, any reason now to use `while (true) { .. }` ?